### ISSUE TITLE
GddStructure missing on SupplementaryPictogram (Bugzilla Bug 441)

### ISSUE THREAD

**Issue by datexii** [DATEX-II-EU/DatexII#441]
![2023_04_26_10_48_08_Window](https://github.com/DATEX-II-EU/DatexII/assets/75827629/320b5134-e8ec-4681-bd0b-f5a956210fea)
This issue was created automatically with bugzilla2github.py
# Bugzilla Bug 441
Date: 2023-04-26T12:04:46+02:00
From: Nicolas Klement &lt;<nicolas.klement@asfinag.at>&gt;
To:   @ingeniumfaber
CC:   dominik.figl@asfinag.at

Last updated: 2023-05-09T11:23:10+02:00

**Comment by AsfFiglDo** [comment]
Any news on that? Will it be fixed in Datex II 4.0?

**Comment by iancornwell1** [comment]
Currently release 4.0 is not planned to update VMS - like with 3.0 it will update Common, LocationReferencing and Situation, with VMS expected to follow in release 4.1 later in 2027. Work on 4.1 has not yet begun, so we have not yet assessed this issue, but it is scheduled to begin this summer.

**Comment by ingeniumfaber** [comment]
This issue need a clarification:

the supplementary pictogram display had been introduced for devices which have a physical, separate device which can display specific supplementary pictograms: when expressing a Pictogram a VMS uses the Pictogram Display and the supplementary display when available

also a unique PIctogram Display have the capability of expressing a Pictogram and also a supplementary pictogram in the unique display pictogram area

the GDD allow to encode a composite Pictogram as a composition of a main pictogram and a supplementary pictogram itself

so  having said that; 

- a supplementary pictogram is managed only when it's a phsysically separated display area dedicated to supplementary pictogram
- a VMS with a Pictogram Display area and no Supplementary Pictogram  Display Area can manage a composite GDD showing a pictogram with supplementary pictogram information 
- when available a Supplementary Pictogram Area physically separated from the Pictogram Display Area the supplementary pictogram is never managed by itself, it is combined with a specific main Pictogram
it should be considered if in such case a GDD can be associated to the single supplementary pictogram area as the GDD allows the whole composite pictogram to be specified and this is the more accurate management of the semantic of pictogram (e.g. slippery road with rain or ice ) which is manage by the composition of main Pictogram and supplementary panel together.

i assume we should clarify such management and allow a better use of GDD in case there is a supplementary physical display panel and a composite GDD should be set on the combination of Pictogram Display Area and Supplementary Display Area. as we need to address one GDD to express such using such composite GDD.

**Comment by ingeniumfaber** [comment]
we diceded to introduce a further association from Supplementary panel to GDD as GDD may also now with new GDD edition convey the supplementary panel by itself. to be done in V4 revision, for version 3 it could be embed in the main GDD pictogram the meaning of the supplementary panel. otherwise an extension will be needed.. GDD 2016 which is referenced in DATEX II v3 do not include supplementary panel definition

**Comment by iancornwell1** [comment]
I agree with the last comment. About the previous comment - “composite” in GDD is for 2 pictograms superimposed in 1 display area, one for exit geometry and one for the nature of the thing at that geometry, and the DATEX II CompositePictogram is also restricted to that purpose due to the values of CompositePictogramEnum. Our CompositePictogram design does not support supplementary display areas in general. That’s why we do need the addition proposed. I checked my thinking with some examples:
•	Example 1 – seems to need model change – main + supplementary pictograms where GDD supplementary pictogram code is used, and the supplementary pictogram is on a dedicated display area. There would be 2 DisplayArea – a PictogramDisplayArea and a SupplementaryPanelArea, and so there would be 2 DisplayAreaSettings  - a PictogramDisplay and a SupplementaryPictogram; the former has a Pictogram with optional attributes and/or GDD code, but the SupplementaryPictogram does not. If we added an optional GddStructure to SupplementaryPictogram, that would allow expression of the GDD information, but it would not allow expression of individual DATEX II numerical attributes for the supplementary pictogram. That might be OK, since you can either express attributes in binary GDD for the SupplementaryPictogram or you can associate the numerical attributes with the the main pictogram that they supplement. We can explain this in Part 4 / documentation. If this is not OK, an even more specific example would be needed to show why. (I could imagine a refactoring of the model to allow numerical attributes directly tied to supplementary pictograms, but it would be added complexity we may not need.)
•	Example 2 – pictogram with supplementary pictogram that can be represented as attributes, all displayed on one panel – GDD encourages the use of attributes and would recommend 1 GDD code; in DATEX there could be one DisplayArea so one DisplayAreaSettings – a PictogramDisplay with 1 Pictogram with 1 GDD code. Attributes can be represented either through attached DisplayedNumericalInformation objects or by GDD binary data.
•	Can we assume that wherever separate main + supplementary pictograms have been used, the DATEX model should reflect 2 display areas in the VmsConfiguration? I hope so.
•	Example 3 – composite pictogram – where GDD has used 2 “InternationalSign”. There is one DisplayArea, so there is one DisplayAreaSettings, and that is a PictogramDisplay with 1 Pictogram, but this can be a CompositePictogram that contains a RegularPictogram, and each has a GDD code, so the GDD information can be represented, albeit in DATEX II we are using two GddStructure with one pictogram code each whereas in GDD there is one GddStructureV2 with 2 InternationalSign. The difference seems not important. 
I had wondered if we would need to also change our multiplicity between GddStructure and GddPictogramIdentification (currently 1 only) but none of these examples show that need.
